### PR TITLE
Added check for skipping files and a force regrid flag.

### PR DESCRIPTION
**Pull request by aniketinfocusp**
This PR includes
- Files are skipped if they are present at output_path
- a `force_regrid` flag to bypass this behavior which is set to False by default.

### CODE REVIEW

File: weather_mv/loader_pipeline/regrid.py [hunk 1]
```diff
@@ -236,6 +239,10 @@ def apply(self, uri: str):
         if self.dry_run:
             return
 
+        if not self.force_regrid and path_exists(self.target_from(uri)):
+            logger.info(f"Skipping {uri}")
```
Comment (mahrsee1997) [inline weather_mv/loader_pipeline/regrid.py:243]: nit: `.`.

Comment (mahrsee1997) [reply weather_mv/loader_pipeline/regrid.py:243]: Also lets add this check for zarr as well i.e. at line [272](https://github.com/google/weather-tools/pull/428/files#diff-7ee93f0ad7c757f82a6e90ea74dc1bdd9c4d04fb1f6c6b9539cd20fe7f2964a6R272).

File: weather_mv/loader_pipeline/sinks.py [hunk 2]
```diff
@@ -380,6 +380,21 @@ def upload(src: str, dst: str) -> None:
     """Uploads a file to the specified GCS bucket destination."""
     subprocess.run(f'gsutil -m cp {src} {dst}'.split(), check=True, capture_output=True, text=True, input="n/n")
 
+def path_exists(path: str) -> bool:
```
Comment (mahrsee1997) [inline weather_mv/loader_pipeline/sinks.py:383]: More cleaner logic: https://github.com/google/weather-tools/blob/main/weather_sp/splitter_pipeline/file_splitters.py#L85C6-L85C6.

**Review by mahrsee1997 — CHANGES_REQUESTED** [review]
Overall looks good ...

File: weather_mv/loader_pipeline/regrid.py (hunk 1 above)
Comment (mahrsee1997) [reply weather_mv/loader_pipeline/regrid.py:243]: Refer to my below comment. Let's move `self.force_regrid` check in skip function.

File: weather_mv/loader_pipeline/sinks.py (hunk 2 above)
Comment (mahrsee1997) [reply weather_mv/loader_pipeline/sinks.py:383]: And for checking in local file system, we can have a local store like: https://github.com/google/weather-tools/blob/main/weather_dl/download_pipeline/stores.py#L80

Comment (aniketinfocusp) [reply weather_mv/loader_pipeline/sinks.py]: updated in 
427038e

File: weather_mv/loader_pipeline/regrid.py (hunk 1 above)
Comment (aniketinfocusp) [reply weather_mv/loader_pipeline/regrid.py]: done

File: weather_mv/loader_pipeline/sinks.py [hunk 3]
```diff
@@ -380,6 +381,24 @@ def upload(src: str, dst: str) -> None:
...
+    """Check if path exists at a certain location. Pass force_regrid to skip checking."""
+    if force_regrid:
+        return False
+    # Try on gcs file system.
+    try:
+        matches = FileSystems().match([path])
+        assert len(matches) == 1
+        if len(matches[0].metadata_list) > 0:
+            return True
+    except Exception as _:
```
Comment (mahrsee1997) [inline weather_mv/loader_pipeline/sinks.py:394]: Let's catch specific exception.

Comment (aniketinfocusp) [reply weather_mv/loader_pipeline/sinks.py]: removed exception

**Review by mahrsee1997 — APPROVED** [review]
LGTM.

**Pull request merged**